### PR TITLE
Adopt Bikeshed best practices and Infra spec conventions

### PR DESCRIPTION
**Pull request by inexorabletash**
A bunch of relatively stand-alone commits addressing issues noted in https://github.com/webmachinelearning/webnn/issues/450

It'll be easiest to review each individual commit as they're each a separate "topic". Reviewers should feel free to cherry-pick changes if anything is deemed controversial.

***

### :boom: Error: 400 Bad Request :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Dec 19, 2023, 6:48 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [CSS Spec Preprocessor](https://api.csswg.org/bikeshed/) - CSS Spec Preprocessor is the web service used to build Bikeshed specs.

:link: [Related URL](https://api.csswg.org/bikeshed/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fwebmachinelearning%2Fwebnn%2Fd5bee458649f3d6f8acf84fa64596d12a66a57e2%2Findex.bs&force=1&md-warning=not%20ready)

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20webmachinelearning/webnn%23495.)._
</details>

### CODE REVIEW

File: index.bs
Comment (zolkis) [inline index.bs:911]: @huningxin do we want to convey more here (and similar places) than just setting the internal slot?

**Comment by zolkis** [comment]
Thanks @inexorabletash, this was tedious work, but it looks much better now.
I am pondering whether we want to capture more in the places where it used to say "Store a reference..." and now say "Set <internal slot>...". But that can be done later as well, so IMHO this PR can be merged as it is, serving as a further baseline. I didn't spot errors, but someone else might also want to take a look.

**Comment by inexorabletash** [comment]
FYI that I'm back from vacation now, and can address any comments from editors etc. Take your time!

**Comment by inexorabletash** [comment]
FYI that I've got another batch of similar fixups coming, but I'll do that as a second PR after this lands, for ease of reviewing.

File: index.bs
Comment (huningxin) [reply index.bs:911]: Just setting the internal slot might be fine. I suppose the graph build algorithm (to be specified) would fetch the platform operator from the internal slot at the graph building stage.

**Review by huningxin — APPROVED** [review]
LGTM, thanks much @inexorabletash !

**Comment by inexorabletash** [comment]
Is anything else needed here? Does this need discussion in the WG meeting?

**Comment by anssiko** [comment]
Thanks for this work. This sets an exemplary baseline to follow. No WG discussion needed for this PR if the editors have no further questions or comments.

@huningxin or @wchao1115 please merge to unblock the next batch of similar improvements.

**Comment by huningxin** [comment]
I have no further questions. We usually need two approvals for a PR merge, I am not sure @wchao1115 has bandwidth on this PR. @anssiko or @zolkis , could you please review and approve? Thanks!

**Review by anssiko — APPROVED** [review]
Well done @inexorabletash! We will note these improvements in the upcoming CR Snapshot. Other implementers watching appreciate these changes.

@zolkis can you resolve your comments to unblock your review? (Or let me know if GH does not let you do it.)

**Review by zolkis — APPROVED** [review]

**Comment by zolkis** [comment]
> @zolkis can you resolve your comments to unblock your review? (Or let me know if GH does not let you do it.)

It was a simple comment, no need to resolve. I gave an explicit approval (even though that's an editors' privilege).
We should merge this.

**Comment by anssiko** [comment]
@huningxin feel free to proceed to merge at your convenience.

**Pull request merged**
